examples/platform_services/android/app/src/main/java/com/example/flutter/ExampleActivity.java

```diff
@@ -36,9 +39,18 @@ public void onMethodCall(MethodCall call, Response response) {
   }
 
   private void getBatteryLevel(Response response) {
-    BatteryManager batteryManager = (BatteryManager) getSystemService(BATTERY_SERVICE);
+    int batteryLevel = -1;
     if (VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP) {
-      response.success(batteryManager.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY));
+      BatteryManager batteryManager = (BatteryManager) getSystemService(BATTERY_SERVICE);
+      batteryLevel = batteryManager.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY);
+    } else {
+      Intent intent = new ContextWrapper(getApplicationContext()).
+          registerReceiver(null, new IntentFilter(Intent.ACTION_BATTERY_CHANGED));
+      batteryLevel = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
+    }
+
+    if (batteryLevel != -1) {
+      response.success(batteryLevel);
     } else {
       response.error("Not available", "Battery level not available.", null);
```

Comment:
[nit] "Not available" could be changed to something more code-like, such as "UNAVAILABLE". Otherwise it looks like we're returning two error messages targeted at the human reader.